src/Native/Unix/System.Native/pal_io.c

```diff
@@ -325,7 +354,30 @@ static void ConvertDirent(const struct dirent* entry, struct DirectoryEntry* out
     // the start of the unmanaged string. Give the caller back a pointer to the
     // location of the start of the string that exists in their own byte buffer.
     outputEntry->Name = entry->d_name;
+#if defined(_AIX)
+    /* AIX has no d_type, make a substitute */
+    struct stat s;
+    stat(entry->d_name, &s);
+    if (S_ISDIR(s.st_mode)) {
```

Comment:
A nit - the coding convention is to place the opening curly bracket at a new line.